javapoet/src/main/java/com/palantir/javapoet/TypeSpec.java

```diff
@@ -284,16 +295,29 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
                 // Push an empty type (specifically without nested types) for type-resolution.
                 codeWriter.pushType(new TypeSpec(this));
 
-                codeWriter.emitJavadoc(javadoc);
+                if (kind == Kind.RECORD && recordConstructor != null) {
```

Comment:
nit: The first condition is redundant

```suggestion
                if (recordConstructor != null) {
```